libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/timeline/RustTimeline.kt

```diff
@@ -293,21 +313,26 @@ class RustTimeline(
         mentions: List<Mention>,
     ): Result<Unit> =
         withContext(dispatcher) {
-            if (originalEventId != null) {
-                runCatching {
-                    val editedEvent = specialModeEventTimelineItem ?: inner.getEventTimelineItemByEventId(originalEventId.value)
-                    editedEvent.use {
-                        inner.edit(
-                            newContent = messageEventContentFromParts(body, htmlBody).withMentions(mentions.map()),
-                            editItem = it,
-                        )
+            runCatching<Unit> {
+                when {
+                    originalEventId != null -> {
+                        val editedEvent = specialModeEventTimelineItem ?: inner.getEventTimelineItemByEventId(originalEventId.value)
+                        editedEvent.use {
+                            inner.edit(
+                                newContent = messageEventContentFromParts(body, htmlBody).withMentions(mentions.map()),
+                                editItem = it,
+                            )
+                        }
+                        specialModeEventTimelineItem = null
```

Comment:
This isn't needed for the other when branch?

Reply:
Actually, we want to remove edition of local echo, as it'll break the ordering in the sending queue.
I'll update the api and usage.